bridge/PendingJobsAsyncWaitTask.h

```diff
@@ -0,0 +1,110 @@
+
+#pragma once
+
+#include "jobmanager/AsyncWaiter.h"
+#include "logger/Logger.h"
+#include "transport/Transport.h"
+#include <functional>
+#include <memory>
+#include <string>
+#include <tuple>
+
+namespace transport
+{
+class Transport;
+}
+
+namespace bridge
+{
+
+template <class T, class... Args>
+class PendingJobsAsyncWaitTask : public jobmanager::AsyncWaitTask
+{
+public:
+    using OnTaskEndHandler = std::function<void()>;
+
+    template <class... UTypes>
+    PendingJobsAsyncWaitTask(const char* loggableId,
+        uint32_t _taskId,
+        T&& pendingJobsChecker,
+        UTypes&&... memoryToHold);
+
+    bool canComplete() final;
+    void onComplete() final;
+    void onTimeout() final;
+
+private:
+    const char* _loggableId;
+    uint32_t _taskId;
+    T _pendingJobsChecker;
+    std::tuple<Args...> _memoryToHold;
+};
+
+template <class T, class... Args>
+template <class... UTypes>
+PendingJobsAsyncWaitTask<T, Args...>::PendingJobsAsyncWaitTask(const char* loggableId,
+    uint32_t _taskId,
+    T&& pendingJobsChecker,
+    UTypes&&... memoryToHold)
+    : _loggableId(loggableId),
+      _taskId(_taskId),
+      _pendingJobsChecker(std::move(pendingJobsChecker)),
+      _memoryToHold(std::forward<UTypes>(memoryToHold)...)
+{
+}
+
+template <class T, class... Args>
+bool PendingJobsAsyncWaitTask<T, Args...>::canComplete()
+{
+    return !_pendingJobsChecker->hasPendingJobs();
+}
+
+template <class T, class... Args>
+void PendingJobsAsyncWaitTask<T, Args...>::onComplete()
+{
+    logger::info("PendingJobsAsyncWaitTask %u has completed", _loggableId, _taskId);
```

Comment:
I do not think we can have this on info level in general. Even debug risk spamming a lot depending on how we use this. Perhaps the logic of the specific wait task has to log depending on scenario.

Reply:
I have implemented a log policy in this commit https://github.com/finos/SymphonyMediaBridge/pull/146/commits/8f7d84718b6a991934d134a916962e30e3bc6e86. Not sure if I am happy with it though.